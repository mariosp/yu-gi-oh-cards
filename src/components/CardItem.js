import React from 'react';
import {Link} from 'react-router-dom';


const CardItem =  (props) =>(

    
    <div>
        <h4>{props.name}</h4>
        <p>{props.type}</p>
    </div>


);

export default CardItem;