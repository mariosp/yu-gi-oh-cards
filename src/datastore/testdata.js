

export default [{
    atk: null,
    card_type: "spell",
    def: null,
    family: null,
    level: null,
    name: "Monster Gate",
    property: "Normal",
    text: "Tribute 1 monster; reveal cards from the top of your Deck until you reveal a monster that can be Normal Summoned. Special Summon it, also send the other revealed cards to the Graveyard.",
    type: null
    
}]