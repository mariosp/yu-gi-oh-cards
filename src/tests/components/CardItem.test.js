import React from 'react'
import {shallow} from 'enzyme';
import CardItem from '../../components/CardItem';
import testdata from '../../datastore/testdata'


test('should render CardItem (sidebar box) with card name and card_type', ()=>{

    const wrapper = shallow(<CardItem name={testdata[0].name} type={testdata[0].card_type} />);
    expect(wrapper).toMatchSnapshot();
    
    
    });






