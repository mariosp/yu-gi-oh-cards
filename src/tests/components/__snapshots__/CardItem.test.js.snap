// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardItem (sidebar box) with card name and card_type 1`] = `
<div
  className="cardbox"
>
  <div
    className="carditem"
  >
    <h5>
      Monster Gate
    </h5>
    <div
      className="ctype"
    >
      spell
    </div>
  </div>
</div>
`;
